src/stores/activeTraceStore.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017 Expedia, Inc.
+ *
+ *       Licensed under the Apache License, Version 2.0 (the "License");
+ *       you may not use this file except in compliance with the License.
+ *       You may obtain a copy of the License at
+ *
+ *           http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *       Unless required by applicable law or agreed to in writing, software
+ *       distributed under the License is distributed on an "AS IS" BASIS,
+ *       WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *       See the License for the specific language governing permissions and
+ *       limitations under the License.
+ *
+ */
+
+import axios from 'axios';
+import {observable, action} from 'mobx';
+import { fromPromise } from 'mobx-utils';
+
+function TraceException(data) {
+    this.message = 'Unable to resolve promise';
+    this.data = data;
+}
+
+function calculateStartTime(spans) {
+    return spans.reduce((earliestTime, span) =>
+        (earliestTime ? Math.min(earliestTime, span.timestamp) : span.timestamp), null
+    );
+}
+
+function calculateDuration(spans, start) {
+    const end = spans.reduce((latestTime, span) =>
+        (latestTime ? Math.max(latestTime, (span.timestamp + span.duration)) : (span.timestamp + span.duration)), null
+    );
+    return (end - start) * 1.05;
+}
+
+
+export class ActiveTraceStore {
+    @observable spans = [];
+    @observable startTime = {};
+    @observable totalDuration = {};
+    @observable promiseState = null;
+    @action fetchTraceDetails(traceId) {
+        this.promiseState = fromPromise(
+            axios
+                .get(`/api/trace/${traceId}`)
+                .then((result) => {
+                    this.spans = result.data;
+                    this.startTime = calculateStartTime(this.spans);
```

Comment:
create a container object `summary` for containing all meta info about the trace eg. startTime, endTime, number  of spans, depth etc.